app/controllers/experiments_controller.rb

```diff
@@ -237,49 +239,25 @@ def move_modal
 
   # POST: move_experiment(id)
   def move
-    project = Project.find_by_id(params[:experiment].try(:[], :project_id))
-    old_project = @experiment.project
-
-    # Try to move the experiment
-    success = true
-    if @experiment.moveable_projects(current_user).include?(project)
-      success = @experiment.move_to_project(project)
-    else
-      success = false
-    end
+    project = Project.find params[:experiment][:project_id]
 
-    if success
-      Activity.create(
-        type_of: :move_experiment,
-        project: project,
-        experiment: @experiment,
-        user: current_user,
-        message: I18n.t(
-          'activities.move_experiment',
-          user: current_user.full_name,
-          experiment: @experiment.name,
-          project_new: project.name,
-          project_original: old_project.name
-        )
-      )
+    service = Experiments::MoveToProjectService
+              .call(experiment_id: @experiment.id,
+                    project_id: project.id,
+                    user_id: current_user.id)
 
+    if service.succeed?
       flash[:success] = t('experiments.move.success_flash',
                           experiment: @experiment.name)
```

Comment:
Use `escape_input` on `@experiment.name`.

Reply:
Not sure how much is needed though.

Reply:
As expected this line dates back when I did some security testing and found a bunch of XSS security issues - 
the [relevant issue](https://biosistemika.atlassian.net/browse/SCI-102?jql=project%20%3D%20SCI%20AND%20text%20~%20%22sci-102%22%20ORDER%20BY%20created%20DESC).

I guess this was then fixed with adding all sorts of [sanitizations](https://github.com/biosistemika/scinote-web/commit/0546716a0b9d81a69fbee45143fb6d577c175d1f) but if I remember correctly the main culprit  was usage of `.html_safe`.  Furthermore not every `@experiment.name` is [sanitized](https://github.com/biosistemika/scinote-web/blame/b26530a02f0020e54807433507f361d395c93976/app/controllers/experiments_controller.rb).

I would assume that this would not be needed if all `html_safe` calls are gone but I don't know the current state of affairs. RoR sanitizes input by default.

Prolly the safest bet is to put it back in without complicating stuff further for this single line, but @Ducz0r can tell if it's redundant.

Reply:
Rails does sanitation of  this by default since version 3. One would have to explicitly request to render non sanitized string with `raw` or `html_safe`.